BEST_PRACTICES.md

```diff
@@ -203,3 +216,459 @@ Use a simple NSURLSession wrapper to make things simpler, [Networking](https://g
+
+Additional capabilities of classes:
+
+- Inheritance: Enables one class to inherit the characteristics of another
+- Type casting: Enables you to check and interpret the type of a class instance at runtime
+- Deinitializers: Enable an instance of a class to free up any resources it has assigned
+- Reference counting: Allows more than one reference to a class instance
+- Compatibility: Classes are available from Objetive-C
+
+### Use of Self
+
+Always use `self` when referencing properties. It will make your life simpler. Trust us.
+
+### Protocol Conformance
+
+When adding protocol conformance to a class, prefer adding a separate class extension for the protocol methods. This keeps the related methods grouped together with the protocol and can simplify instructions to add a protocol to a class with its associated methods. Also don't use extensions to split code, but to extend classes or structs with additional functionality. Finally, avoid the `// MARK: -` it makes code too noisy.
+
+**Preferred:**
+```swift
+class MyViewcontroller: UIViewController {
+    // class stuff here
+}
+
+extension MyViewcontroller: UITableViewDataSource {
+    // table view data source methods
+}
+
+extension MyViewcontroller: UIScrollViewDelegate {
+    // scroll view delegate methods
+}
+```
+
+**Not Preferred:**
+```swift
+class MyViewcontroller: UIViewController, UITableViewDataSource, UIScrollViewDelegate {
+    // all methods
+}
+
+// MARK: - Data download
+extension MyViewController {
+
+}
+```
+
+### Computed Properties
+
+For conciseness, if a computed property is read-only, omit the get clause. The get clause is required only when a set clause is provided.
+
+**Preferred:**
+```swift
+var diameter: Double {
+    return radius * 2.0
+}
+```
+
+**Not Preferred:**
+```swift
+var diameter: Double {
+    get {
+      return radius * 2.0
+    }
+}
+```
+
+### Example definition
+
+Here's an example of a well-styled class definition:
+
+```swift
+class Circle: Shape {
+    var x: Int
+    var y: Int
+    var radius: Double
+
+    var diameter: Double {
+        get {
+            return self.radius * 2.0
+        }
+        set {
+            radius = newValue / 2.0
+        }
+    }
+
+    init(x: Int, y: Int, radius: Double) {
+        self.x = x
+        self.y = y
+        self.radius = radius
+    }
+
+    convenience init(x: Int, y: Int, diameter: Double) {
+        self.init(x: x, y: y, radius: diameter / 2.0)
+    }
+
+    func describe() -> String {
+        return "I am a circle at \(centerString()) with an area of \(computeArea())"
+    }
+
+    override func computeArea() -> Double {
+        return M_PI * self.radius * self.radius
+    }
+
+    private func centerString() -> String {
+        return "(\(x),\(y))"
+    }
+}
+```
+
+The example above demonstrates the following style guidelines:
+
+ + The correct spacing for variable assignations is with a space after and before the equals mark `=`, e.g. `x = 3`
```

Comment:
I think it's implied. Or not? A space, doesn't mean a double space or a triple space, just one space.